File: linked_li.py
class Node:
    __slots__='_element','_next'
    def __init__(self,element,next):
        self._element=element
        self._next=next


class LinkedList:
    def __init__(self):
        self._head=None
        self._tail=None
        self._size=0
    
    def __len__(self):
        return self._size
    
    def isEmpty(self):
        return self._size==0
    
    def addLast(self,e):
        newest=Node(e,None)
        if self.isEmpty():
            self._head=newest
            # print("head",newest)
        else:
            self._tail._next=newest
            # print("tail",self._tail._next)
        self._tail=newest
        self._size+=1
    
    def display(self):
        p=self._head
        while p:
            print(p._element,end="--->")
            p=p._next
        print()
# searching a element in the linked list....
    def Search(self,key):
        p=self._head
        i=0
        while p:
            if p._element==key:
                return i
            else:
                i=i+1
                p=p._next
        return "element not found..."

# insert the element at the first.
    def addFirst(self,val):
        newest=Node(val,None)
        if self.isEmpty():
            self._head=newest
            self._tail=newest
            
        else:
            newest._next=self._head
            self._head=newest
        self._size+=1

    
# insert element at any where in the list........
    def insertElement(self,e,pos):
        newest=Node(e,None)
        p=self._head
        i=0
        while i<(pos-1):
            p=p._next
            i=i+1
        newest._next=p._next
        p._next=newest
        self._size+=1

    # delete the element at the starting.
    def delFirst(self):
        self._head=self._head._next
    
    # delete the element in any where.
    def removeany(self, position):
        p = self._head
        i = 1
        while i < position - 1:
            p = p._next
            i = i + 1
        e = p._next._element
        p._next = p._next._next
        self._size -= 1
        return e


L=LinkedList()
L.addLast(4)
L.addLast(7)
L.addLast(72)
L.addLast(75)
L.addLast(71)
L.display()
L.addFirst(44)
L.display()
print(len(L))
L.insertElement(99,3)
L.display()
print(L.Search(88))
print(len(L))
# L.delFirst()
# L.delFirst()

print(L.removeany(3))
L.display()

File: linked_li.go
package main

import (
	"fmt"
)

type Node struct {
	value int
	next  *Node
}

type LinkedList struct {
	head *Node
	tail *Node
	len  int
}

func (l *LinkedList) Insert(val int) {
	n := Node{}
	n.value = val

	if l.head == nil {
		l.head = &n
	} else {
		l.tail.next = &n
	}
	l.tail = &n
	l.len++
	return
}

func (l *LinkedList) addFirst(val int) {
	newNode := Node{value: val}
	if l.head == nil {
		l.head = &newNode
		l.tail = &newNode
	} else {
		newNode.next = l.head
		l.head = &newNode
	}
	l.len++

}

func (l *LinkedList) insertAny(val, pos int) {
	newNode := &Node{value: val}
	ptr := l.head
	for i := 1; i < pos-1; i++ {
		ptr = ptr.next
	}
	newNode.next = ptr.next
	ptr.next = newNode
	l.len++

}

func (l *LinkedList) display() {
	if l.len == 0 {
		fmt.Println("linked list is empty...")
	} else {
		ptr := l.head
		for i := 0; i < l.len; i++ {
			fmt.Printf("%d-->", ptr.value)
			ptr = ptr.next
		}
		fmt.Println()

	}
}

func (l *LinkedList) delFirst() {
	// fmt.Println(l.head)
	// fmt.Println((l.head.next))
	l.head = (l.head.next)
	fmt.Println(l.head)
	l.len--
}
func (l *LinkedList) delAny(pos int) {
	p := l.head
	for i := 1; i < pos-1; i++ {
		p = p.next
	}
	p.next = p.next.next
	l.len--
}

func (l *LinkedList) delLast() {
	p := l.head
	for i := 1; i < l.len-1; i++ {
		p = p.next
		// fmt.Println(p)
	}
	p.next = nil
	l.len--

}

func main() {
	l := LinkedList{}
	l.Insert(78)
	l.Insert(87)
	l.Insert(90)
	l.display()
	// fmt.Println(strings.Repeat("#", 20))
	// l.addFirst(34)
	// l.display()
	// fmt.Println(strings.Repeat("#", 20))
	l.insertAny(33, 2)
	l.display()
	// fmt.Println(strings.Repeat("#", 20))
	// l.delFirst()
	// l.display()
	// fmt.Println(strings.Repeat("#", 20))
	// // l.delAny(3)
	// l.delLast()
	l.display()
	fmt.Println("the length of list is:", l.len)

}
